os/include/semaphore.h

```diff
@@ -79,13 +79,21 @@
 #define PRIOINHERIT_FLAGS_DISABLE (1 << 0) /* Bit 0: Priority inheritance
 					    * is disabled for this semaphore */
 #define FLAGS_INITIALIZED         (1 << 1) /* Bit 1: This semaphore initialized */
+#ifdef CONFIG_BINMGR_RECOVERY
+#define FLAGS_ZEROSEM             (1 << 2) /* Bit 2: The semaphore for signaling */
+#endif
+
+/* Save semaphore holder data with enabling priority inheritance or binary manager recovery */
```

Comment:
I think
`Save semaphore holder data when priority inheritance or binary manager recovery is enabled.`
is better.